contracts/JoepegAuctionHouse.sol

```diff
@@ -0,0 +1,960 @@
+        address indexed creator,
+        address indexed collection,
+        uint256 indexed tokenId
+    );
+
+    event NewCurrencyManager(address indexed currencyManager);
+    event NewEnglishAuctionMinBidIncrementPct(
+        uint256 englishAuctionMinBidIncrementPct
+    );
+    event NewEnglishAuctionRefreshTime(uint256 englishAuctionRefreshTime);
+    event NewProtocolFeeManager(address indexed protocolFeeManager);
+    event NewProtocolFeeRecipient(address indexed protocolFeeRecipient);
+    event NewRoyaltyFeeManager(address indexed royaltyFeeManager);
+
+    event RoyaltyPayment(
+        address indexed collection,
+        uint256 indexed tokenId,
+        address indexed royaltyRecipient,
+        address currency,
+        uint256 amount
+    );
+
+    modifier isSupportedCurrency(address _currency) {
+        if (!currencyManager.isCurrencyWhitelisted(_currency)) {
+            revert JoepegAuctionHouse__UnsupportedCurrency();
+        } else {
+            _;
+        }
+    }
+
+    ///  @notice Initializer
+    ///  @param _englishAuctionMinBidIncrementPct minimum bid increment percentage for English Auctions
+    ///  @param _englishAuctionRefreshTime refresh time for English auctions
+    ///  @param _currencyManager currency manager address
+    ///  @param _protocolFeeManager protocol fee manager address
+    ///  @param _royaltyFeeManager royalty fee manager address
+    ///  @param _wavax address of WAVAX
+    ///  @param _protocolFeeRecipient protocol fee recipient
+    function initialize(
+        uint256 _englishAuctionMinBidIncrementPct,
+        uint256 _englishAuctionRefreshTime,
+        address _currencyManager,
+        address _protocolFeeManager,
+        address _royaltyFeeManager,
+        address _wavax,
+        address _protocolFeeRecipient
+    ) public initializer {
+        if (_englishAuctionMinBidIncrementPct > PERCENTAGE_PRECISION) {
+            revert JoepegAuctionHouse__EnglishAuctionInvalidMinBidIncrementPct();
+        }
+
+        __Ownable_init();
+
+        englishAuctionMinBidIncrementPct = _englishAuctionMinBidIncrementPct;
+        englishAuctionRefreshTime = _englishAuctionRefreshTime;
+        currencyManager = ICurrencyManager(_currencyManager);
+        protocolFeeManager = IProtocolFeeManager(_protocolFeeManager);
+        royaltyFeeManager = IRoyaltyFeeManager(_royaltyFeeManager);
+        protocolFeeRecipient = _protocolFeeRecipient;
+        WAVAX = _wavax;
+    }
+
+    /// @notice Starts an English Auction for an ERC721 token
+    /// @dev Note this requires the auction house to hold the ERC721 token in escrow
+    /// @param _collection address of ERC721 token
+    /// @param _tokenId token id of ERC721 token
+    /// @param _currency address of currency to sell ERC721 token for
+    /// @param _duration number of seconds for English Auction to run
+    /// @param _startPrice minimum starting bid price
+    function startEnglishAuction(
+        address _collection,
```

Comment:
I kept it as `address` for some callsites b/c I was using it as `address` multiple times in the function body but only needed to convert to `IERC721` once so thought its cleaner this way.
slightly tangential to this, do you think I should add a check in the beginning of the function to ensure that the collection conforms to ERC721 interface (i.e. isn't ERC1155 token)? I believe since I use `IERC721.safeTransferFrom` it will revert if it isnt ERC721 but that happens at the end of the function

Reply:
makes sense, thanks for the thorough explanation! i've changed them to `IERC721` type in the params